RedHat-UX/red-hat-design-system
# feat(dialog): use native HTML dialog element

## What I did

1. This is a continuation of #1865. Opening this as a separate PR because of merge conflicts in 1865.
2. Please review the PR body and comments thoroughly from 1865 for changes
    * Highlights:
        * Now uses the native HTML `<dialog>` element.
        * Added the `accessible-label` property
        * Improved modal styles
        * Focus trapping
        * Updated a11y docs
        * rh-surface wrapping only `<dialog>` element

## Testing Instructions

1. Visit [the DP Demo](https://deploy-preview-2078--red-hat-design-system.netlify.app/elements/dialog/demo/).
2. View the other demos:
    * [Color Context](https://deploy-preview-2078--red-hat-design-system.netlify.app/elements/dialog/demo/color-context/)
    * [Events](https://deploy-preview-2078--red-hat-design-system.netlify.app/elements/dialog/demo/events/)
    * [Lots of Content](https://deploy-preview-2078--red-hat-design-system.netlify.app/elements/dialog/demo/lots-of-content)
    * [No headings](https://deploy-preview-2078--red-hat-design-system.netlify.app/elements/dialog/demo/no-headings/)
    * [No header content](https://deploy-preview-2078--red-hat-design-system.netlify.app/elements/dialog/demo/no-header-content/)
    * [Video](https://deploy-preview-2078--red-hat-design-system.netlify.app/elements/dialog/demo/video-modal/)
3. Add various elements to modal header, body, footer
4. Test variants (`variant="small"` etc and `position="top"`)
5. Check a11y
6. Check for appropriate spacing

## Notes to Reviewers

### Docs updates:

  - [x] Some docs on the Guidelines page are incorrect as they have the modal body [scrolling independently](http://localhost:8080/elements/dialog/style/#configuration) from the header and footer. This is not possible short of a variant.
  - [x] Guidelines for the [video player close button location](http://localhost:8080/elements/dialog/guidelines/#video-player-dialog) differ from actual implementation (update image)
  - [x] Guidelines for [Overflow content](http://localhost:8080/elements/dialog/guidelines/#overflow-content) differ from implementation (update text, image)
  - [x] Border radius of dialog in images

Closes https://github.com/RedHat-UX/red-hat-design-system/issues/1242 and https://github.com/RedHat-UX/red-hat-design-system/issues/1238.
